terraform/modules/cds_export/policy.tf

```diff
@@ -0,0 +1,37 @@
+data "aws_iam_policy_document" "cds_assume_role" {
+  statement {
+    actions = [
+      "sts:AssumeRole"
+    ]
+
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = [var.cds_access_role_arn]
+    }
+  }
+}
+
+#tfsec:ignore:aws-iam-no-policy-wildcards:wildcard required to allow access to all files at the root of the bucket
+data "aws_iam_policy_document" "export_bucket_read_only_access" {
+  statement {
+    actions = [
+      "s3:Get*",
+      "s3:List*",
+      "s3:Describe*"
+    ]
+
+    effect = "Allow"
+
+    resources = [
+      aws_s3_bucket.this.arn,
+      "${aws_s3_bucket.this.arn}/*"

```

Comment:
I think this is necessary to allow access to the bucket and all the files at the root of it (i.e. the csv's and zip files)

Reply:
That wildcard is, yes.

I think overall this policy can be more restrictive however - e.g. I don't think they need to be able to retrieve all of the bucket config information, just list the bucket contents and be able to retrieve any object in the bucket. (This probably involves two statements in this policy to separate actions allowed on the bucket from actions allowed on the contents of the bucket)

As it's read access it's not a huge problem, but it is worth looking briefly at restricting it further.

Reply:
Think I've managed to resolve this if you can sanity check it 😊